pkg/machinery/resources/network/dns_resolve_cache.go

```diff
@@ -0,0 +1,65 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package network
+
+import (
+	"net/netip"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/protobuf"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+
+	"github.com/siderolabs/talos/pkg/machinery/proto"
+)
+
+// DNSResolveCacheType is type of DNSResolveCache resource.
+const DNSResolveCacheType = resource.Type("DNSServers.net.talos.dev")
```

Comment:
Should the type name stay unchanged (`DNSServers`)?

Reply:
it should be updated to match the constant